x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectProcessManager.java

```diff
@@ -133,6 +139,15 @@ public AutodetectProcessManager(Environment environment, Settings settings, Clie
         this.jobResultsPersister = jobResultsPersister;
         this.jobDataCountsPersister = jobDataCountsPersister;
         this.auditor = auditor;
+        this.nativeStorageProvider = new NativeStorageProvider(environment);
+    }
+
+    public void onNodeStartup() {
+        try {
+            nativeStorageProvider.cleanupLocalTmpStorageInCaseOfUncleanShutdown();
+        } catch (IOException e) {
```

Comment:
It might be worth catching `Exception` here as a runtime exception creeping in in the future might prevent the node from starting and it's probably not a good reason to do so.